Makefile

```diff
@@ -242,6 +242,35 @@ cli-test-s3-latest:
 	./opstrace --help
 
 
+.PHONY: check-license-headers
+check-license-headers:
+	@echo "does addlicense work?"
+	command -v addlicense
+	# Walk through the directory tree. For every directory (that is not
+	# ./.git), execute a bash process. Echo the current directory. Set the
+	# extended globbing option. Add a newline in the command; for that option
+	# to take effect (Otherwise bash would show a syntax error on line 0). For
+	# being able to add a newline in a single-quoted string we have to add a
+	# dollar sign to the front of the string (See
+	# https://stackoverflow.com/a/3182519/145400). To make that work in this
+	# Makefile, escape the single dollar sign with another dollar sign. Run
+	# `addlicense` so that it _modifies_ the source files. `addlicense` will
+	# show a "no such file or directory" error when the extended glob does not
+	# match a single file in the current directory. That's fine. Note: we start
+	# bash so that we can rely on bash's extended glob behavior. Also note that
+	# `addlicense` does not have well-behaved recursive directory tree walking
+	# behavior (with predictable pattern matching!). That's why we use
+	# find/bash+extglob to control _that_ part. Fail the target when this
+	# technique produces file changes (show the changes, too).
+	find . -not -path '*/.git/*' \
+		-not -path '*/node_modules*' \
+		-not -path '*/kubernetes/src/custom-resources*' \
+		-type d -exec bash -c \
+		$$'echo "dir: {}"; shopt -s extglob \n addlicense -c "Opstrace, Inc." -l apache {}/@(*.ts|*.tsx|*.json|*.go|*.css)' \;
```

Comment:
Note: add new file extensions here, when they come up.

Reply:
Also note: the `{}` is where `find` fills in the path to the current direction (as part of the file system walk).

Reply:
Also note: launching a constant number of processes process per directory (bash and addlicense) is probably faster here than launching a constant number of processes per file, assuming that we probably have ~10 times more files than directories.

In CI, this walk passes within ~1 second. On my machine, it's a little slower (5 seconds or so).

Reply:
CC @triclambert not sure if you're interested in seeing this ugly piece of code, but here it is :shrug:

Reply:
I've been following along.  And steering clear.  😁   

But in all seriousness, thanks for doing this.  It's unfortunate that it's needed, but your solution looks good.